src/manifolds/ProbabilitySimplex.jl

```diff
@@ -0,0 +1,294 @@
+
+"""
+    check_manifold_point(M::ProbabilitySimplex, p; kwargs...)
+
+Check whether `p` is a valid point on the [`ProbabilitySimplex`](@ref) `M`, i.e. is a point in
+the embedding with positive entries that sum to one
+The tolerance for the last test can be set using the `kwargs...`.
+"""
+function check_manifold_point(M::ProbabilitySimplex, p; kwargs...)
+    mpv = invoke(
+        check_manifold_point,
+        Tuple{(typeof(get_embedding(M))), typeof(p)},
+        get_embedding(M),
+        p;
+        kwargs...
+    )
+    mpv === nothing || return mpv
+    if any(p .<= 0)
+        return DomainError(
+            norm(p),
+            "The point $(p) does not lie on the $(M) since it has nonpositive entries.",
+        )
+    end
+    if !isapprox(sum(p), 1.0; kwargs...)
+        return DomainError(
+            sum(p),
+            "The point $(p) does not lie on the $(M) since its sum is not 1.",
+        )
+    end
+    return nothing
+end
+
+"""
+    check_tangent_vector(M::ProbabilitySimplex, p, X; check_base_point = true, kwargs... )
+
+Check whether `X` is a tangent vector to `p` on the [`ProbabilitySimplex`](@ref) `M`, i.e.
+after [`check_manifold_point`](@ref)`(M,p)`, `X` has to be of same dimension as `p`
+and its elements have to sum to one.
+The optional parameter `check_base_point` indicates, whether to call
+[`check_manifold_point`](@ref)  for `p` or not.
+The tolerance for the last test can be set using the `kwargs...`.
+"""
+function check_tangent_vector(
+    M::ProbabilitySimplex,
+    p,
+    X;
+    check_base_point = true,
+    kwargs...,
+)
+    if check_base_point
+        mpe = check_manifold_point(M, p; kwargs...)
+        mpe === nothing || return mpe
+    end
+    mpv = invoke(
+        check_tangent_vector,
+        Tuple{typeof(get_embedding(M)), typeof(p), typeof(X)},
+        get_embedding(M),
+        p,
+        X;
+        check_base_point = false, # already checked above
+        kwargs...
+    )
+    mpv === nothing || return mpv
+    if !isapprox(sum(X), 0.0; kwargs...)
+        return DomainError(
+            sum(X),
+            "The vector $(X) is not a tangent vector to $(p) on $(M), since its elements to not sum up to 0.",
+        )
+    end
+    return nothing
+end
+
+decorated_manifold(M::ProbabilitySimplex) = Euclidean(representation_size(M)...; field = ℝ)
+
+default_metric_dispatch(::ProbabilitySimplex, ::FisherRaoMetric) = Val(true)
+
+@doc raw"""
+    distance(M,p,q)
+
+Compute the distance between two points on the [`ProbabilitySimplex`](@ref) `M`.
+The formula reads
+````math
+d_{Δ^n}(p,q) = 2\arccos \biggl( \sum_{i=1}^{n+1} \sqrt{p_i q_i} \biggr)
+````
+"""
+function distance(::ProbabilitySimplex, p, q)
+    sumsqrt = zero(Base.promote_eltype(p, q))
+    @inbounds for i in eachindex(p, q)
+        sumsqrt += sqrt(p[i] * q[i])
+    end
+    return 2*acos(sumsqrt)
+end
+
+embed!(::ProbabilitySimplex, q, p) = copyto!(q, p)
+embed!(::ProbabilitySimplex, Y, p, X) = copyto!(Y, X)
```

Comment:
This, I think, depends on whether we want an isometric embedding or not. This one is not isometric so scaling is not necessary (although I guess we may prefer an isometric embedding as a default for `ProbabilitySimplex`).

Reply:
Currently it's the `DefaultEmbedding`, that does not enforce (or use) any property. If we scale correctly we could go to `IsometricEmbedding´ and not implement inner – but use the embedding fallback for that. I feel that here our inner is faster than the default.

I currently use the embedding only within the checks, since I can just pass `p` or `X` to the embedding to check its size and that it's real valued (since that's all that `Euclidean` checks)

So the Isometric one might be nice to have, but we have to overwrite inner anyways, so there wouldn't be much use in terms of lines of code.

I personally have to preference, since the checks I did would be invariant to the scale